src/ui.form-date-picker-range/index.vue

```diff
@@ -885,7 +885,67 @@ function getPeriodButtonsClasses(periodName) {
   return period.value === periodName ? config.value.periodButtonActive : "";
 }
 
-function getPeriodDateClasses(date) {
+function getPeriodDateClasses(date, index) {
+  const localStart = new Date(localValue.value.from);
+  const localEnd = new Date(localValue.value.to);
+  const isListType = isPeriod.value.quarter || isPeriod.value.week;
+  const firstInRangeClasses = isListType
+    ? config.value.firstPeriodListDate
+    : config.value.firstPeriodGridDate;
+  const lastInRangeClasses = isListType
+    ? config.value.lastPeriodListDate
+    : config.value.lastPeriodGridDate;
+
+  if (isPeriod.value.year) {
+    localStart.setMonth(0, 1);
+    localEnd.setMonth(11, 31);
+  }
+
+  localStart.setHours(0, 0, 0, 0);
+  localEnd.setHours(23, 59, 59, 999);
+
+  const startDateInRangeIndex = periodDateList.value.findIndex((periodDate) => {
+    return localStart <= periodDate.endRange && localStart >= periodDate.startRange;
+  });
+
+  const endDateInRangeIndex = periodDateList.value.findIndex((periodDate) => {
+    return localEnd >= periodDate.startRange && localEnd <= periodDate.endRange;
+  });
+
+  const isStartDateInRangeIndex = startDateInRangeIndex !== -1;
+  const isEndDateInRangeIndex = endDateInRangeIndex !== -1;
```

Comment:
use ~ and const no needed

Reply:
Its still better to use const and keep result rather than call those operators each time.

Reply:
Maybe but it not hevy operation. So I'm not sure this optimisation is really needed. You just comparing 2 values.

Reply:
It's not significant but does no harm and code is more readable this way, imo.